source/read/change-streams.txt

```diff
@@ -0,0 +1,278 @@
+             },
+             "removedFields": [],
+             "truncatedArrays": []
+           }
+         }
+       ],
+       ...
+     }
+   }
+
+Modify the Change Stream Output
+-------------------------------
+
+You can pass the ``pipeline`` parameter to the ``watch()`` method to modify the
+change stream output. This parameter allows you to watch for only specified
+change events. Format the parameter as a list of objects that each represent an
+aggregation stage.
+
+You can specify the following stages in the ``pipeline`` parameter:
+
+- ``$addFields``
+- ``$match``
+- ``$project``
+- ``$replaceRoot``
+- ``$replaceWith``
+- ``$redact``
+- ``$set``
+- ``$unset``
+
+The following example uses the ``pipeline`` parameter to open a change stream
+that records only update operations:
+
+.. literalinclude:: /includes/read/change-streams.kt
+   :start-after: start-change-stream-pipeline
+   :end-before: end-change-stream-pipeline
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+To learn more about modifying your change stream output, see the
+:manual:`Modify Change Stream Output
+</changeStreams/#modify-change-stream-output>` section in the {+mdb-server+}
+manual.
+
+Modify ``watch()`` Behavior
+---------------------------
+
+The ``watch()`` method can be modified by chaining methods to the ``ChangeStreamIterable``
+object returned by the ``watch()`` method call. If you don't specify any options, the
+driver does not customize the operation.
+
+The following table describes methods you can use to customize the behavior
+of ``watch()``:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``batchSize()``
+     - | Sets the number of documents to return per batch.
+
+   * - ``collation()``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
+
+   * - ``comment()``
+     - | Specifies a comment to attach to the operation.
+
+   * - ``forEach()``
+     - | Performs the given action on each element and safely closes the cursor.
+
+   * - ``fullDocument()``
+     - | Sets the ``fullDocument`` value. To learn more, see the
+         :ref:`<kotlin-sync-change-stream-pre-post-image>` section of this document.
+
+   * - ``fullDocumentBeforeChange()``
+     - | Sets the ``fullDocumentBeforeChange`` value. To learn more, see the
+         :ref:`<kotlin-sync-change-stream-pre-post-image>` section of this document.
+
+   * - ``maxAwaitTime()``
+     - | Sets the maximum await execution time on the server for this operation, in
+         milliseconds.
+
+   * - ``toCollection()``
+     - | Appends all elements to the given destination collection.
+
+For a complete list of methods you can use to configure the ``watch()`` method, see
+the `ChangeStreamIterable <{+api+}/com.mongodb.kotlin.client/-change-stream-iterable/index.html>`__
```

Comment:
Note for reviewer: these links are broken at the moment.

Rea has a change that has yet to be merged that changes the `{+api+}` source constant. This will work once it's merged.